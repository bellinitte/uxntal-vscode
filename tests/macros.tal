%MACRO { }
%! { }
%@ { }
%% { }
% { }

( Braces )
%MACRO { }
%MACRO{ }
%MACRO{x }
%MACRO{x }x
%MACRO { { { }
%MACRO { } } }
%MACRO { { } }

%MACRO { x %NESTED { x } x }

( Macro contents )
%MACRO {
    BRK ADD2kr @label/label &sublabel
    .label #12 #abcd $1 |100 ~lib.tal
    "string %MACRO ( comment ) [ ]
}
