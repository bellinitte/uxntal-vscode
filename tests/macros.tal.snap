>%MACRO { }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.braces.uxntal
>%! { }
#^ source.tal punctuation.rune.uxntal
# ^ source.tal meta.preprocessor.uxntal
#  ^ source.tal
#   ^ source.tal punctuation.braces.uxntal
#    ^ source.tal
#     ^ source.tal punctuation.braces.uxntal
>%@ { }
#^ source.tal punctuation.rune.uxntal
# ^ source.tal meta.preprocessor.uxntal
#  ^ source.tal
#   ^ source.tal punctuation.braces.uxntal
#    ^ source.tal
#     ^ source.tal punctuation.braces.uxntal
>%% { }
#^ source.tal punctuation.rune.uxntal
# ^ source.tal meta.preprocessor.uxntal
#  ^ source.tal
#   ^ source.tal punctuation.braces.uxntal
#    ^ source.tal
#     ^ source.tal punctuation.braces.uxntal
>% { }
#^ source.tal punctuation.rune.uxntal
# ^ source.tal
#  ^ source.tal punctuation.braces.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.braces.uxntal
>
>( Braces )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^ source.tal comment.uxntal
#        ^ source.tal
#         ^ source.tal comment.uxntal
>%MACRO { }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.braces.uxntal
>%MACRO{ }
#^ source.tal punctuation.rune.uxntal
# ^^^^^^ source.tal meta.preprocessor.uxntal
#       ^ source.tal
#        ^ source.tal punctuation.braces.uxntal
>%MACRO{x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^^^ source.tal meta.preprocessor.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.braces.uxntal
>%MACRO{x }x
#^ source.tal punctuation.rune.uxntal
# ^^^^^^^ source.tal meta.preprocessor.uxntal
#        ^ source.tal
#         ^^ source.tal punctuation.braces.uxntal
>%MACRO { { { }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.braces.uxntal
#          ^ source.tal
#           ^ source.tal punctuation.braces.uxntal
#            ^ source.tal
#             ^ source.tal punctuation.braces.uxntal
>%MACRO { } } }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.braces.uxntal
#          ^ source.tal
#           ^ source.tal punctuation.braces.uxntal
#            ^ source.tal
#             ^ source.tal punctuation.braces.uxntal
>%MACRO { { } }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.braces.uxntal
#          ^ source.tal
#           ^ source.tal punctuation.braces.uxntal
#            ^ source.tal
#             ^ source.tal punctuation.braces.uxntal
>
>%MACRO { x %NESTED { x } x }
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
#        ^ source.tal
#         ^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal punctuation.rune.uxntal
#            ^^^^^^ source.tal meta.preprocessor.uxntal
#                  ^ source.tal
#                   ^ source.tal punctuation.braces.uxntal
#                    ^ source.tal
#                     ^ source.tal entity.name.function.uxntal
#                      ^ source.tal
#                       ^ source.tal punctuation.braces.uxntal
#                        ^ source.tal
#                         ^ source.tal entity.name.function.uxntal
#                          ^ source.tal
#                           ^ source.tal punctuation.braces.uxntal
>
>( Macro contents )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^ source.tal comment.uxntal
#       ^ source.tal
#        ^^^^^^^^ source.tal comment.uxntal
#                ^ source.tal
#                 ^ source.tal comment.uxntal
>%MACRO {
#^ source.tal punctuation.rune.uxntal
# ^^^^^ source.tal meta.preprocessor.uxntal
#      ^ source.tal
#       ^ source.tal punctuation.braces.uxntal
>    BRK LIT2kr @label/label &sublabel
#^^^^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^^^^ source.tal keyword.uxntal
#              ^ source.tal
#               ^ source.tal punctuation.rune.uxntal
#                ^^^^^ source.tal entity.name.tag.uxntal
#                     ^ source.tal punctuation.rune.uxntal
#                      ^^^^^ source.tal entity.name.tag.uxntal
#                           ^ source.tal
#                            ^ source.tal punctuation.rune.uxntal
#                             ^^^^^^^^ source.tal entity.name.tag.uxntal
>    .label #12 #abcd $1 |100 ~lib.tal
#^^^^ source.tal
#    ^ source.tal punctuation.rune.uxntal
#     ^^^^^ source.tal entity.name.function.uxntal
#          ^ source.tal
#           ^ source.tal punctuation.rune.uxntal
#            ^^ source.tal constant.numeric.uxntal
#              ^ source.tal
#               ^ source.tal punctuation.rune.uxntal
#                ^^^^ source.tal constant.numeric.uxntal
#                    ^ source.tal
#                     ^ source.tal punctuation.rune.uxntal
#                      ^ source.tal constant.numeric.uxntal
#                       ^ source.tal
#                        ^ source.tal punctuation.rune.uxntal
#                         ^^^ source.tal constant.numeric.uxntal
#                            ^ source.tal
#                             ^ source.tal punctuation.rune.uxntal
#                              ^^^^^^^ source.tal meta.preprocessor.string.uxntal
>    "string %MACRO ( comment ) [ ]
#^^^^ source.tal
#    ^ source.tal punctuation.rune.uxntal
#     ^^^^^^ source.tal string.uxntal
#           ^ source.tal
#            ^ source.tal punctuation.rune.uxntal
#             ^^^^^ source.tal meta.preprocessor.uxntal
#                  ^ source.tal
#                   ^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^^^^^^^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^ source.tal comment.uxntal
#                              ^ source.tal
#                               ^ source.tal punctuation.brackets.uxntal
#                                ^ source.tal
#                                 ^ source.tal punctuation.brackets.uxntal
>}
#^ source.tal punctuation.braces.uxntal
>